lib/models/entity.ex

```diff
@@ -3,17 +3,43 @@ defmodule ExMicrosoftBot.Models.Entity do
   Microsoft bot entity structure
   """
 
+  alias ExMicrosoftBot.Models.Entity.Mention
+
   @derive [Poison.Encoder]
-  defstruct [:type, :name, :supportsDisplay]
+  defstruct [
+    :type,
+
+    # type == "?"
+    :name,
+    :supportsDisplay,
+
+    # type == "clientInfo"
+    :country,
+    :locale,
+    :platform,
+    :timezone,
+
+    # type == "mention"
+    :mentioned,
+    :text
+  ]
 
   @type t :: %ExMicrosoftBot.Models.Entity{
```

Comment:
@jmnsf how about the following? Though it won't make it more strongly typed but I think makes it more clear what the various shapes of Entity could be

```
  @type t :: %ExMicrosoftBot.Models.Entity{
          type: String.t(),
          name: String.t() | nil,
          supportsDisplay: boolean() | nil,
        } | %ExMicrosoftBot.Models.Entity{
          type: String.t(), # clientInfo
          country: String.t() | nil,
          locale: String.t() | nil,
          platform: String.t() | nil,
          timezone: String.t() | nil,
        } | %ExMicrosoftBot.Models.Entity{
          type: String.t(), # mention
          mentioned: Mention.t() | nil,
          text: String.t() | nil
        }
```

Reply:
@zabirauf done 👍🏼. Unfortunate that we can't use string literals here, but added some docs that'll hopefully make things clearer.